meilisearch-http/src/extractors/authentication/mod.rs

```diff
@@ -50,11 +50,17 @@ impl<P, D> GuardedData<P, D> {
     {
         match Self::authenticate(auth, String::new(), None).await? {
             Some(filters) => match data {
-                Some(data) => Ok(Self {
-                    data,
-                    filters,
-                    _marker: PhantomData,
-                }),
+                Some(data) => {
+                    if filters.is_missing_master_key() {
+                        Err(AuthenticationError::MissingMasterKey.into())
+                    } else {
+                        Ok(Self {
+                            data,
+                            filters,
+                            _marker: PhantomData,
+                        })
+                    }
+                }
                 None => Err(AuthenticationError::IrretrievableState.into()),
             },
             None => Err(AuthenticationError::MissingAuthorizationHeader.into()),
```

Comment:
Honestly, I'd prefer something like:
```suggestion
            None =>  match auth.get_master_key() {
                Some(_) => Err(AuthenticationError::MissingAuthorizationHeader.into()),
                None => Err(AuthenticationError::MissingMasterKey.into()),
            }
```
it seems simpler 🤔

Reply:
Made the change as suggested. Thanks!